### ISSUE TITLE
dig - DNS over TLS Invalid argument error

### ISSUE THREAD

**Issue by tomakey** [Microsoft/WSL#3089]
* Your Windows build number:  Microsoft Windows [Version 10.0.14393]

* What you're doing and what's happening:  
Running dig over TLS (port 853) produces this error:
`dig yahoo.com @9.9.9.9 -p 853`
`socket.c:1915: internal_send: 9.9.9.9#853: Invalid argument`

* What's wrong / what should be happening instead:
Running regular dig over default port is fine:
`dig yahoo.com @9.9.9.9`

; <<>> DiG 9.9.5-3ubuntu0.17-Ubuntu <<>> yahoo.com @9.9.9.9
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 37226
;; flags: qr rd ra; QUERY: 1, ANSWER: 6, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;yahoo.com.                     IN      A

;; ANSWER SECTION:
yahoo.com.              721     IN      A       98.138.219.232
yahoo.com.              721     IN      A       72.30.35.9
yahoo.com.              721     IN      A       98.137.246.7
yahoo.com.              721     IN      A       98.137.246.8
yahoo.com.              721     IN      A       98.138.219.231
yahoo.com.              721     IN      A       72.30.35.10

;; Query time: 62 msec
;; SERVER: 9.9.9.9#53(9.9.9.9)
;; WHEN: Tue Apr 10 22:25:59 DST 2018
;; MSG SIZE  rcvd: 134

* Strace of the failing command, if applicable: 
https://gist.github.com/tomakey/3ffc4bfd6fe379ca0603f63e844808b9

See [our contributing instructions](https://github.com/Microsoft/BashOnWindows/blob/master/CONTRIBUTING.md) for assistance.

**Comment by therealkenc** [comment]
A few fails here.
```
590   setsockopt(20, SOL_SOCKET, SO_TIMESTAMP, [1], 4) = 0
590   setsockopt(20, SOL_IP, IP_MTU_DISCOVER, [0], 4) = -1 EINVAL (Invalid argument)
[...]
590   sendmsg(20, {msg_name(16)={sa_family=AF_INET, sin_port=htons(853), sin_addr=inet_addr("9.9.9.9")}, msg_iov(1)=[{"4a\1 \0\1\0\0\0\0\0\1\5yahoo\3com\0\0\1\0\1\0\0)\20\0"..., 38}], msg_controllen=0, msg_flags=0}, 0 <unfinished ...>
590   <... sendmsg resumed> )           = -1 EINVAL (Invalid argument)
[...]
589   open("/proc/sys/vm/overcommit_memory", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
589   madvise(0x7f5fcc021000, 45056, MADV_DONTNEED) = 0
```
We've got `/proc/sys/vm/overcommit_memory` these days. I think that `EINVAL` is the fatal. That will be some dupe or another but it's hard to search with all the networking noise in this hub.

Get bind for Windows [here](https://www.isc.org/downloads/).

**Comment by tomakey** [comment]
Thanks fpqc - not really an option here; enterprise computers are slower to adopt newer releases.

@therealkenc - windows binary works; also tried to compile 9.11.3 in WSL Ubuntu - failed with sightly different errors. I guess I'll let this go for now. 
`socket.c:2938: setsockopt(20, IP_RECVTOS) failed: Invalid argument`
`socket.c:2135: internal_send: 9.9.9.9#853: Invalid argument`

Gist for Dig 9.11.3:
https://gist.github.com/04d4f4e7753d64902db26734e37db559

**Comment by therealkenc** [comment]
Thanks. I just tried on 17639 and `dig` appears to work on Ubuntu, at least to a first order. No `EINVAL`, and the lookup works (I'll spare the screenshot). Don't know when it was fixed. Closing fixedincreatorsupdate (a total unsupported guess). This should be marked as a dupe but I don't have a reference (and I'm having a low motivation day to look harder). If you ever manage to upgrade to something recent and still see problems, ping this issue and we can revisit.
